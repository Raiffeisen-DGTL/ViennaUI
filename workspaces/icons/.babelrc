{
  "presets": ["@babel/env", "@babel/typescript", "@babel/react"],
  "env": {
    "esm": {
      "presets": [["@babel/env", { "modules": false }]]
    }
  }
}
