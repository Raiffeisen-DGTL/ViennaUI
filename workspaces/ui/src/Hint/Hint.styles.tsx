import styled from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';

const definition = getPresets('hint.definition', {
    base: null,
    custom: null,
});

const wrapper = getPresets('hint.wrapper', {
    base: null,
    custom: null,
});

const icon = getPresets('hint.icon', {
    base: null,
    hover: null,
    custom: null,
    size: 'size',
});

export const Definition = styled.span`
    ${definition.base}
    ${definition.custom}
`;

export const Wrapper = styled.span`
    ${wrapper.base}
    ${wrapper.custom}
`;

export const Icon = styled.svg<{ size?: 's' | 'm' | 'l' }>`
    ${icon.base}
    ${icon.size}
    &:hover {
        ${icon.hover}
    }
    &:focus {
        ${icon.hover}
    }
    ${icon.custom}
`;
