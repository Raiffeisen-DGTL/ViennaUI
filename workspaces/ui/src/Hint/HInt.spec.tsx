import React from 'react';
import { Close } from 'vienna.icons';
import { Hint, HintProps } from './Hint';
import { Icon } from './Hint.styles';
import { Popover } from '../Popover';
import { Content, Message } from '../Popover/Popover.styles';

test('Hint', () => {
    const snap = snapshot.render(
        <Hint
            design='dark'
            anchor='bottom'
            header='This is header'
            content='Ut ut dui non ipsum pharetra aliquam id in libero. Aenean placerat leo nec ex pharetra finibus. Nunc id turpis eu risus rhoncus cursus quis efficitur sapien. Donec vitae arcu at lectus placerat fringilla.'
        />
    );

    expect(snap).toMatchSnapshot();
});

test('Hint w/ design', () => {
    const designs: HintProps['design'][] = ['dark', 'light'];

    designs.forEach((design) => {
        const snap = snapshot.render(
            <Hint
                design={design}
                anchor='bottom'
                header='This is header'
                content='Ut ut dui non ipsum pharetra aliquam id in libero. Aenean placerat leo nec ex pharetra finibus. Nunc id turpis eu risus rhoncus cursus quis efficitur sapien. Donec vitae arcu at lectus placerat fringilla.'
            />
        );

        expect(snap).toMatchSnapshot();
    });
});

test('Hint w/ size', () => {
    const sizes: HintProps['size'][] = ['s', 'm', 'l'];

    sizes.forEach((size) => {
        const snap = snapshot.render(
            <Hint
                size={size}
                anchor='bottom'
                header='This is header'
                content='Ut ut dui non ipsum pharetra aliquam id in libero. Aenean placerat leo nec ex pharetra finibus. Nunc id turpis eu risus rhoncus cursus quis efficitur sapien. Donec vitae arcu at lectus placerat fringilla.'
            />
        );

        expect(snap).toMatchSnapshot();
    });
});

test('Hint /w focus', () => {
    const snap = snapshot.mount(
        <Hint
            design='dark'
            anchor='bottom'
            header='This is header'
            content='Ut ut dui non ipsum pharetra aliquam id in libero. Aenean placerat leo nec ex pharetra finibus. Nunc id turpis eu risus rhoncus cursus quis efficitur sapien. Donec vitae arcu at lectus placerat fringilla.'
        />
    );

    const icon = snap.find(Icon);
    icon.simulate('focus');
    const content = snap.find(Content);

    expect(content.exists()).toEqual(true);
});

test('Hint /w close', () => {
    const snap = snapshot.mount(
        <Hint
            design='dark'
            anchor='bottom'
            header='This is header'
            content='Ut ut dui non ipsum pharetra aliquam id in libero. Aenean placerat leo nec ex pharetra finibus. Nunc id turpis eu risus rhoncus cursus quis efficitur sapien. Donec vitae arcu at lectus placerat fringilla.'
        />
    );

    const icon = snap.find(Icon);
    icon.simulate('focus');

    const content = snap.find(Content);

    expect(content.exists()).toEqual(true);

    const close = snap.find(Close);
    close.simulate('click');

    const message = snap.find(Message);
    expect(message.props().toggle).toEqual(false);
});

test('Hint w/ any props', () => {
    const props = {
        id: 'id',
        title: 'title',
    };

    const component = <Hint {...props} />;

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();

    const wrapper = snapshot.mount(component);
    const popover = wrapper.find(Popover);
    for (const [key, value] of Object.entries(props)) {
        expect(popover.props()[key]).toEqual(value);
    }
});
