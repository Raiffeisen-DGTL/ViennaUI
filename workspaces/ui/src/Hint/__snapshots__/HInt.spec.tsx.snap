// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hint 1`] = `
.c0 {
  outline: none;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: sub;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  opacity: 0;
  width: 250px;
  min-width: 250px;
  box-shadow: 0 8px 32px 0 rgba(0,0,0,0.08);
  border-radius: 8px;
  padding-top: 12px;
  padding-right: 28px;
  padding-bottom: 12px;
  padding-left: 12px;
  z-index: 1000;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
  background-color: #2B2D33;
  color: #FFFFFF;
  top: 22px;
  left: 0px;
  right: px;
  bottom: px;
  display: none;
}

.c4 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
  font-weight: 700;
  margin-bottom: 4px;
}

.c5 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
}

.c3 {
  position: absolute;
  top: 12px;
  right: 12px;
  cursor: pointer;
}

.c1 {
  color: #AAABAD;
  outline: none;
  cursor: pointer;
  border-radius: 16px;
  width: 16px;
  height: 16px;
}

.c1:hover {
  color: #2B2D33;
}

.c1:focus {
  color: #2B2D33;
}

<span
  class="c0"
  tabindex="0"
>
  <svg
    class="c1"
    fill="none"
    height="18"
    tabindex="1"
    viewBox="0 0 18 18"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 0C13.9706 0 18 4.02944 18 9C18 13.9706 13.9706 18 9 18C4.02944 18 0 13.9706 0 9C0 4.02944 4.02944 0 9 0ZM9 7.2H7.2V9H8.1V11.7H6.3V13.5H11.7V11.7H9.9V8.1C9.9 7.60294 9.49706 7.2 9 7.2ZM9 3.6C8.25441 3.6 7.65 4.20442 7.65 4.95C7.65 5.69558 8.25441 6.3 9 6.3C9.74558 6.3 10.35 5.69558 10.35 4.95C10.35 4.20442 9.74558 3.6 9 3.6Z"
      fill="currentcolor"
    />
  </svg>
  <div
    class="c2"
    width="250"
  >
    <div
      class="c3"
    >
      <svg
        fill="currentColor"
        height="16"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
        />
      </svg>
    </div>
    <div
      class="c4"
    >
      This is header
    </div>
    <div
      class="c5"
    >
      Ut ut dui non ipsum pharetra aliquam id in libero. Aenean placerat leo nec ex pharetra finibus. Nunc id turpis eu risus rhoncus cursus quis efficitur sapien. Donec vitae arcu at lectus placerat fringilla.
    </div>
  </div>
</span>
`;

exports[`Hint w/ any props 1`] = `
.c0 {
  outline: none;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: sub;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  opacity: 0;
  width: 250px;
  min-width: 250px;
  box-shadow: 0 8px 32px 0 rgba(0,0,0,0.08);
  border-radius: 8px;
  padding-top: 12px;
  padding-right: 28px;
  padding-bottom: 12px;
  padding-left: 12px;
  z-index: 1000;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
  background-color: #FFFFFF;
  color: #2B2D33;
  top: 22px;
  left: 0px;
  right: px;
  bottom: px;
  display: none;
}

.c4 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
}

.c3 {
  position: absolute;
  top: 12px;
  right: 12px;
  cursor: pointer;
}

.c1 {
  color: #AAABAD;
  outline: none;
  cursor: pointer;
  border-radius: 16px;
  width: 16px;
  height: 16px;
}

.c1:hover {
  color: #2B2D33;
}

.c1:focus {
  color: #2B2D33;
}

<span
  class="c0"
  id="id"
  tabindex="0"
  title="title"
>
  <svg
    class="c1"
    fill="none"
    height="18"
    tabindex="1"
    viewBox="0 0 18 18"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 0C13.9706 0 18 4.02944 18 9C18 13.9706 13.9706 18 9 18C4.02944 18 0 13.9706 0 9C0 4.02944 4.02944 0 9 0ZM9 7.2H7.2V9H8.1V11.7H6.3V13.5H11.7V11.7H9.9V8.1C9.9 7.60294 9.49706 7.2 9 7.2ZM9 3.6C8.25441 3.6 7.65 4.20442 7.65 4.95C7.65 5.69558 8.25441 6.3 9 6.3C9.74558 6.3 10.35 5.69558 10.35 4.95C10.35 4.20442 9.74558 3.6 9 3.6Z"
      fill="currentcolor"
    />
  </svg>
  <div
    class="c2"
    width="250"
  >
    <div
      class="c3"
    >
      <svg
        fill="currentColor"
        height="16"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
        />
      </svg>
    </div>
    <div
      class="c4"
    />
  </div>
</span>
`;

exports[`Hint w/ design 1`] = `
.c0 {
  outline: none;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: sub;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  opacity: 0;
  width: 250px;
  min-width: 250px;
  box-shadow: 0 8px 32px 0 rgba(0,0,0,0.08);
  border-radius: 8px;
  padding-top: 12px;
  padding-right: 28px;
  padding-bottom: 12px;
  padding-left: 12px;
  z-index: 1000;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
  background-color: #2B2D33;
  color: #FFFFFF;
  top: 22px;
  left: 0px;
  right: px;
  bottom: px;
  display: none;
}

.c4 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
  font-weight: 700;
  margin-bottom: 4px;
}

.c5 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
}

.c3 {
  position: absolute;
  top: 12px;
  right: 12px;
  cursor: pointer;
}

.c1 {
  color: #AAABAD;
  outline: none;
  cursor: pointer;
  border-radius: 16px;
  width: 16px;
  height: 16px;
}

.c1:hover {
  color: #2B2D33;
}

.c1:focus {
  color: #2B2D33;
}

<span
  class="c0"
  tabindex="0"
>
  <svg
    class="c1"
    fill="none"
    height="18"
    tabindex="1"
    viewBox="0 0 18 18"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 0C13.9706 0 18 4.02944 18 9C18 13.9706 13.9706 18 9 18C4.02944 18 0 13.9706 0 9C0 4.02944 4.02944 0 9 0ZM9 7.2H7.2V9H8.1V11.7H6.3V13.5H11.7V11.7H9.9V8.1C9.9 7.60294 9.49706 7.2 9 7.2ZM9 3.6C8.25441 3.6 7.65 4.20442 7.65 4.95C7.65 5.69558 8.25441 6.3 9 6.3C9.74558 6.3 10.35 5.69558 10.35 4.95C10.35 4.20442 9.74558 3.6 9 3.6Z"
      fill="currentcolor"
    />
  </svg>
  <div
    class="c2"
    width="250"
  >
    <div
      class="c3"
    >
      <svg
        fill="currentColor"
        height="16"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
        />
      </svg>
    </div>
    <div
      class="c4"
    >
      This is header
    </div>
    <div
      class="c5"
    >
      Ut ut dui non ipsum pharetra aliquam id in libero. Aenean placerat leo nec ex pharetra finibus. Nunc id turpis eu risus rhoncus cursus quis efficitur sapien. Donec vitae arcu at lectus placerat fringilla.
    </div>
  </div>
</span>
`;

exports[`Hint w/ design 2`] = `
.c0 {
  outline: none;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: sub;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  opacity: 0;
  width: 250px;
  min-width: 250px;
  box-shadow: 0 8px 32px 0 rgba(0,0,0,0.08);
  border-radius: 8px;
  padding-top: 12px;
  padding-right: 28px;
  padding-bottom: 12px;
  padding-left: 12px;
  z-index: 1000;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
  background-color: #FFFFFF;
  color: #2B2D33;
  top: 22px;
  left: 0px;
  right: px;
  bottom: px;
  display: none;
}

.c4 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
  font-weight: 700;
  margin-bottom: 4px;
}

.c5 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
}

.c3 {
  position: absolute;
  top: 12px;
  right: 12px;
  cursor: pointer;
}

.c1 {
  color: #AAABAD;
  outline: none;
  cursor: pointer;
  border-radius: 16px;
  width: 16px;
  height: 16px;
}

.c1:hover {
  color: #2B2D33;
}

.c1:focus {
  color: #2B2D33;
}

<span
  class="c0"
  tabindex="0"
>
  <svg
    class="c1"
    fill="none"
    height="18"
    tabindex="1"
    viewBox="0 0 18 18"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 0C13.9706 0 18 4.02944 18 9C18 13.9706 13.9706 18 9 18C4.02944 18 0 13.9706 0 9C0 4.02944 4.02944 0 9 0ZM9 7.2H7.2V9H8.1V11.7H6.3V13.5H11.7V11.7H9.9V8.1C9.9 7.60294 9.49706 7.2 9 7.2ZM9 3.6C8.25441 3.6 7.65 4.20442 7.65 4.95C7.65 5.69558 8.25441 6.3 9 6.3C9.74558 6.3 10.35 5.69558 10.35 4.95C10.35 4.20442 9.74558 3.6 9 3.6Z"
      fill="currentcolor"
    />
  </svg>
  <div
    class="c2"
    width="250"
  >
    <div
      class="c3"
    >
      <svg
        fill="currentColor"
        height="16"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
        />
      </svg>
    </div>
    <div
      class="c4"
    >
      This is header
    </div>
    <div
      class="c5"
    >
      Ut ut dui non ipsum pharetra aliquam id in libero. Aenean placerat leo nec ex pharetra finibus. Nunc id turpis eu risus rhoncus cursus quis efficitur sapien. Donec vitae arcu at lectus placerat fringilla.
    </div>
  </div>
</span>
`;

exports[`Hint w/ size 1`] = `
.c0 {
  outline: none;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: sub;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  opacity: 0;
  width: 250px;
  min-width: 250px;
  box-shadow: 0 8px 32px 0 rgba(0,0,0,0.08);
  border-radius: 8px;
  padding-top: 12px;
  padding-right: 28px;
  padding-bottom: 12px;
  padding-left: 12px;
  z-index: 1000;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
  background-color: #FFFFFF;
  color: #2B2D33;
  top: 22px;
  left: 0px;
  right: px;
  bottom: px;
  display: none;
}

.c4 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
  font-weight: 700;
  margin-bottom: 4px;
}

.c5 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
}

.c3 {
  position: absolute;
  top: 12px;
  right: 12px;
  cursor: pointer;
}

.c1 {
  color: #AAABAD;
  outline: none;
  cursor: pointer;
  border-radius: 16px;
  width: 12px;
  height: 12px;
}

.c1:hover {
  color: #2B2D33;
}

.c1:focus {
  color: #2B2D33;
}

<span
  class="c0"
  tabindex="0"
>
  <svg
    class="c1"
    fill="none"
    height="18"
    tabindex="1"
    viewBox="0 0 18 18"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 0C13.9706 0 18 4.02944 18 9C18 13.9706 13.9706 18 9 18C4.02944 18 0 13.9706 0 9C0 4.02944 4.02944 0 9 0ZM9 7.2H7.2V9H8.1V11.7H6.3V13.5H11.7V11.7H9.9V8.1C9.9 7.60294 9.49706 7.2 9 7.2ZM9 3.6C8.25441 3.6 7.65 4.20442 7.65 4.95C7.65 5.69558 8.25441 6.3 9 6.3C9.74558 6.3 10.35 5.69558 10.35 4.95C10.35 4.20442 9.74558 3.6 9 3.6Z"
      fill="currentcolor"
    />
  </svg>
  <div
    class="c2"
    width="250"
  >
    <div
      class="c3"
    >
      <svg
        fill="currentColor"
        height="16"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
        />
      </svg>
    </div>
    <div
      class="c4"
    >
      This is header
    </div>
    <div
      class="c5"
    >
      Ut ut dui non ipsum pharetra aliquam id in libero. Aenean placerat leo nec ex pharetra finibus. Nunc id turpis eu risus rhoncus cursus quis efficitur sapien. Donec vitae arcu at lectus placerat fringilla.
    </div>
  </div>
</span>
`;

exports[`Hint w/ size 2`] = `
.c0 {
  outline: none;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: sub;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  opacity: 0;
  width: 250px;
  min-width: 250px;
  box-shadow: 0 8px 32px 0 rgba(0,0,0,0.08);
  border-radius: 8px;
  padding-top: 12px;
  padding-right: 28px;
  padding-bottom: 12px;
  padding-left: 12px;
  z-index: 1000;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
  background-color: #FFFFFF;
  color: #2B2D33;
  top: 22px;
  left: 0px;
  right: px;
  bottom: px;
  display: none;
}

.c4 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
  font-weight: 700;
  margin-bottom: 4px;
}

.c5 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
}

.c3 {
  position: absolute;
  top: 12px;
  right: 12px;
  cursor: pointer;
}

.c1 {
  color: #AAABAD;
  outline: none;
  cursor: pointer;
  border-radius: 16px;
  width: 16px;
  height: 16px;
}

.c1:hover {
  color: #2B2D33;
}

.c1:focus {
  color: #2B2D33;
}

<span
  class="c0"
  tabindex="0"
>
  <svg
    class="c1"
    fill="none"
    height="18"
    tabindex="1"
    viewBox="0 0 18 18"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 0C13.9706 0 18 4.02944 18 9C18 13.9706 13.9706 18 9 18C4.02944 18 0 13.9706 0 9C0 4.02944 4.02944 0 9 0ZM9 7.2H7.2V9H8.1V11.7H6.3V13.5H11.7V11.7H9.9V8.1C9.9 7.60294 9.49706 7.2 9 7.2ZM9 3.6C8.25441 3.6 7.65 4.20442 7.65 4.95C7.65 5.69558 8.25441 6.3 9 6.3C9.74558 6.3 10.35 5.69558 10.35 4.95C10.35 4.20442 9.74558 3.6 9 3.6Z"
      fill="currentcolor"
    />
  </svg>
  <div
    class="c2"
    width="250"
  >
    <div
      class="c3"
    >
      <svg
        fill="currentColor"
        height="16"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
        />
      </svg>
    </div>
    <div
      class="c4"
    >
      This is header
    </div>
    <div
      class="c5"
    >
      Ut ut dui non ipsum pharetra aliquam id in libero. Aenean placerat leo nec ex pharetra finibus. Nunc id turpis eu risus rhoncus cursus quis efficitur sapien. Donec vitae arcu at lectus placerat fringilla.
    </div>
  </div>
</span>
`;

exports[`Hint w/ size 3`] = `
.c0 {
  outline: none;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  vertical-align: sub;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  opacity: 0;
  width: 250px;
  min-width: 250px;
  box-shadow: 0 8px 32px 0 rgba(0,0,0,0.08);
  border-radius: 8px;
  padding-top: 12px;
  padding-right: 28px;
  padding-bottom: 12px;
  padding-left: 12px;
  z-index: 1000;
  -webkit-transition: opacity .2s;
  transition: opacity .2s;
  background-color: #FFFFFF;
  color: #2B2D33;
  top: 22px;
  left: 0px;
  right: px;
  bottom: px;
  display: none;
}

.c4 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
  font-weight: 700;
  margin-bottom: 4px;
}

.c5 {
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-stretch: normal;
  font-style: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  margin: 0;
  font-size: 12px;
  line-height: 18px;
}

.c3 {
  position: absolute;
  top: 12px;
  right: 12px;
  cursor: pointer;
}

.c1 {
  color: #AAABAD;
  outline: none;
  cursor: pointer;
  border-radius: 16px;
  width: 20px;
  height: 20px;
}

.c1:hover {
  color: #2B2D33;
}

.c1:focus {
  color: #2B2D33;
}

<span
  class="c0"
  tabindex="0"
>
  <svg
    class="c1"
    fill="none"
    height="18"
    tabindex="1"
    viewBox="0 0 18 18"
    width="18"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 0C13.9706 0 18 4.02944 18 9C18 13.9706 13.9706 18 9 18C4.02944 18 0 13.9706 0 9C0 4.02944 4.02944 0 9 0ZM9 7.2H7.2V9H8.1V11.7H6.3V13.5H11.7V11.7H9.9V8.1C9.9 7.60294 9.49706 7.2 9 7.2ZM9 3.6C8.25441 3.6 7.65 4.20442 7.65 4.95C7.65 5.69558 8.25441 6.3 9 6.3C9.74558 6.3 10.35 5.69558 10.35 4.95C10.35 4.20442 9.74558 3.6 9 3.6Z"
      fill="currentcolor"
    />
  </svg>
  <div
    class="c2"
    width="250"
  >
    <div
      class="c3"
    >
      <svg
        fill="currentColor"
        height="16"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs />
        <path
          d="M12 13.414l-7.293 7.293-1.414-1.414L10.586 12 3.293 4.707l1.414-1.414L12 10.586l7.293-7.293 1.414 1.414L13.414 12l7.293 7.293-1.414 1.414L12 13.414z"
        />
      </svg>
    </div>
    <div
      class="c4"
    >
      This is header
    </div>
    <div
      class="c5"
    >
      Ut ut dui non ipsum pharetra aliquam id in libero. Aenean placerat leo nec ex pharetra finibus. Nunc id turpis eu risus rhoncus cursus quis efficitur sapien. Donec vitae arcu at lectus placerat fringilla.
    </div>
  </div>
</span>
`;
